src/fluent-bit.c

```diff
@@ -1396,12 +1392,18 @@ int flb_main(int argc, char **argv)
 #ifdef FLB_SYSTEM_WINDOWS
         flb_console_handler_set_ctx(ctx, cf_opts);
 #endif
+        if (flb_bin_restarting == FLB_RELOAD_IN_PROGRESS) {
+            /* reload by using same config files/path */
+            flb_reload(ctx, cf_opts);
+            ctx = flb_context_get();
+            flb_bin_restarting = FLB_RELOAD_IDLE;
+        }
     }
 
     if (exit_signal) {
         flb_signal_exit(exit_signal);
     }
-    ret = config->exit_status_code;
+    ret = ctx->config->exit_status_code;
```

Comment:
This is to prevent touching old config and following warning.

```
^C[2023/10/14 10:28:56] [engine] caught signal (SIGINT)
==33368== Invalid read of size 4
==33368==    at 0x1C84B2: flb_main (fluent-bit.c:1404)
==33368==    by 0x1C859B: main (fluent-bit.c:1437)
==33368==  Address 0x541f6d8 is 808 bytes inside a block of size 18,592 free'd
==33368==    at 0x484B27F: free (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==33368==    by 0x22418D: flb_free (flb_mem.h:127)
==33368==    by 0x22C28B: flb_config_exit (flb_config.c:545)
==33368==    by 0x1DDB0A: flb_destroy (flb_lib.c:242)
==33368==    by 0x28ECFB: flb_reload (flb_reload.c:504)
==33368==    by 0x1C6E86: flb_signal_handler (fluent-bit.c:613)
==33368==    by 0x4FA151F: ??? (in /usr/lib/x86_64-linux-gnu/libc.so.6)
==33368==    by 0x50447F7: clock_nanosleep@@GLIBC_2.17 (clock_nanosleep.c:78)
==33368==    by 0x5049676: nanosleep (nanosleep.c:25)
==33368==    by 0x50495AD: sleep (sleep.c:55)
==33368==    by 0x1C846A: flb_main (fluent-bit.c:1380)
==33368==    by 0x1C859B: main (fluent-bit.c:1437)
==33368==  Block was alloc'd at
==33368==    at 0x484DA83: calloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==33368==    by 0x224173: flb_calloc (flb_mem.h:95)
==33368==    by 0x22B647: flb_config_init (flb_config.c:201)
==33368==    by 0x1DD88C: flb_create (flb_lib.c:162)
==33368==    by 0x1C784D: flb_main (fluent-bit.c:1050)
==33368==    by 0x1C859B: main (fluent-bit.c:1437)
==33368== 
```